manifests/tftp.pp

```diff
@@ -13,11 +13,22 @@
     recurse => true;
   }
 
-  foreman_proxy::tftp::sync_file{$foreman_proxy::tftp_syslinux_files:
-    source_path => $foreman_proxy::tftp_syslinux_root,
+  # TODO: remove on the next major version bump
+  if $foreman_proxy::tftp_syslinux_files {
+    foreman_proxy::tftp::sync_file{$foreman_proxy::tftp_syslinux_files:
+      source_path => $foreman_proxy::tftp_syslinux_root,
+      target_path => $foreman_proxy::tftp_root,
+      require     => Class['tftp::install'];
+    }
+
+    warning('foreman_proxy::tftp_syslinux_files is deprecated and will be removed')
```

Comment:
oh, and please add a note to the init.pp parameter documentation to say the old parameters are deprecated.